contracts/interfaces/draft-IERC4337.sol

```diff
@@ -0,0 +1,268 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity ^0.8.20;
+
+/**
+ * @dev An user operation that can be executed by an account on its packed version.
+ *
+ * - `sender`: The account making the operation (might not be deployed yet).
+ * - `nonce`: Anti-replay parameter handled by the entrypoint.
+ * - `initCode`: Concatenation of the address of the factory and the factory calldata.
+ * - `callData`: The `data` the sender will be called with.
+ * - `accountGasLimits`: Concatenation of the verification step gas limit and the main call gas limit.
+ * - `preVerificationGas`: Extra gas to pay the bundler before the verification step.
+ * - `gasFees`: Maximum fee and priority fee the sender is willing to pay.
+ * - `paymasterAndData`: Address of paymaster contract and calldata, (or empty, if account pays for itself).
+ * - `signature`: Data passed into the account to verify authorization.
```

Comment:
This section as been modified from the account-abstration branch.

The account abstraction branch contains the full description of the content of a user operation, with the exact same comments as in the ERC: https://github.com/ethereum/ercs/blob/master/ERCS/erc-4337.md#useroperation

IMO it would be better to keep all the list of all fields in a (not packed) user operation for documentation ... an only keep the packing description bellow